plugins/woocommerce/src/Admin/RemoteInboxNotifications/OptionRuleProcessor.php

```diff
@@ -23,9 +23,33 @@ public function process( $rule, $stored_state ) {
 		$is_contains   = $rule->operation && strpos( $rule->operation, 'contains' ) !== false;
 		$default_value = $is_contains ? array() : false;
 		$default       = isset( $rule->default ) ? $rule->default : $default_value;
-		$option_value  = get_option( $rule->option_name, $default );
+		$option_value  = $this->get_option_value( $rule, $default, $is_contains );
 
-		if ( $is_contains && ! is_array( $option_value ) ) {
+		if ( isset( $rule->transformers ) && is_array( $rule->transformers ) ) {
+			$option_value = TransformerService::apply( $option_value, $rule->transformers, $default );
+		}
+
+		return ComparisonOperation::compare(
+			$option_value,
+			$rule->value,
+			$rule->operation
+		);
+	}
+
+	/**
+	 * Retrieves the option value and handles logging if necessary.
+	 *
+	 * @param object $rule         The specific rule being processed.
+	 * @param mixed  $default      The default value.
+	 * @param bool   $is_contains  Indicates whether the operation is "contains".
+	 *
+	 * @return mixed The option value.
+	 */
+	private function get_option_value( $rule, $default, $is_contains ) {
+		$option_value      = get_option( $rule->option_name, $default );
+		$is_contains_valid = $is_contains && ( is_array( $option_value ) || ( is_string( $option_value ) && is_string( $rule->value ) ) );
+
+		if ( ! $is_contains_valid ) {
```

Comment:
In order for this logic to work correctly you did have to add `$is_contains &&` to this condition as well. As any other condition outside of `$is_contains` would render `$is_contains_valid` as false, causing all the option values being overwritten to an empty array.

Reply:
Good call! I fixed it in the commit [a71ac47](https://github.com/woocommerce/woocommerce/pull/38312/commits/a71ac474c7c1e8efdb3c1a9464ab08f90e8ca231)